internal/oci/uvm.go

```diff
@@ -332,11 +340,19 @@ func SpecToUVMCreateOpts(ctx context.Context, s *specs.Spec, id, owner string) (
 		case uvm.PreferredRootFSTypeVHD:
 			lopts.RootFSFile = uvm.VhdFile
 		}
+		lopts.VPCIEnabled = parseAnnotationsBool(ctx, s.Annotations, annotationVPCIEnabled, lopts.VPCIEnabled)
+		lopts.KernelDirect = parseAnnotationsBool(ctx, s.Annotations, annotationKernelDirectBoot, lopts.KernelDirect)
+		if !lopts.KernelDirect {
+			lopts.KernelFile = uvm.KernelFile
+		}
 		lopts.BootFilesPath = parseAnnotationsString(s.Annotations, annotationBootFilesRootPath, lopts.BootFilesPath)
 		return lopts, nil
 	} else if IsWCOW(s) {
 		wopts := uvm.NewDefaultOptionsWCOW(id, owner)
 		wopts.MemorySizeInMB = ParseAnnotationsMemory(ctx, s, annotationMemorySizeInMB, wopts.MemorySizeInMB)
+		wopts.LowMmioGapInMB = parseAnnotationsUint64(ctx, s.Annotations, annotationMemoryLowMmioGapInMB, wopts.LowMmioGapInMB)
+		wopts.HighMmioBaseInMB = parseAnnotationsUint64(ctx, s.Annotations, annotationMemoryHighMmioGapInMB, wopts.HighMmioBaseInMB)
```

Comment:
It looks like you have the annotations for `HighMmioGap` and `HighMmioBase` flipped here.